src/psqlgraph/hydrator.py

```diff
@@ -509,57 +514,117 @@ def is_parent_relation(self, label, relation):
 
-        link_found = False
-        for assoc_name, assoc_meta in src_node._pg_edges.items():
-            if isinstance(dst_node, assoc_meta["type"]):
-                getattr(src_node, assoc_name).append(dst_node)
-                link_found = True
-                break
-
-        if not link_found:
-            logging.debug(
-                "Could not find a direct relation between '{}'<->'{}'".format(
-                    src_node.label, dst_node.label
+        if association_name:
+            getattr(src_node, association_name).append(dst_node)
+            return
+
+        # attempt to get association by going through all links defined on the source node if needed.
+        association_names = self.get_association_by_edge_name(src_node, dst_node, edge_label)
+
+        if len(association_names) == 1:
+            getattr(src_node, association_names[0]).append(dst_node)
+
+        if len(association_names) > 1:
+            if strict:
+                raise PSQLGraphError(
+                    f"Multiple associations '{association_names}' found between '{src_node.label}' and '{dst_node.label}'"
+                )
+            logger.warning(
+                "Multiple association '%s' found between '%s' and '%s'",
+                association_names,
+                src_node.label,
+                dst_node.label,
+            )
+        if len(association_names) == 0:
+            if strict:
+                raise PSQLGraphError(
+                    f"Could not find a direct relation (edge name or label = '{edge_label}') between '{src_node.label}' and '{dst_node.label}' "
                 )
+            logger.warning(
+                "Could not find a direct relation (edge name or label = '%s') between '%s' and '%s'",
+                edge_label,
+                src_node.label,
+                dst_node.label,
             )
+
+        if len(association_names) == 0:
+            # try the reverse
+            self.make_association(dst_node, src_node, edge_label, strict=True)
+
+    def get_association_by_edge_name(
+        self, src_node: psqlgraph.Node, dst_node: psqlgraph.Node, edge_name: Optional[str] = None
+    ) -> List[str]:
+        """Get the association name used to link the src and dst nodes
+        Args:
+            src_node: the source node
+            dst_node: the destination node
+            edge_name: the edge name as defined in the dictionary.
+                If None, a unique association between the two nodes will be used.
+                An exception is raised if no unique association is found.
+        Returns:
+            The name of the edge from the source node.
+            For example, if the source node is case and the destination
+            node is aliquot, the name of the edge from the `case` node is `aliquots`
+        """
+        association_names = []
+        for assoc_name, assoc_meta in src_node._pg_edges.items():  # noqa
```

Comment:
I do not think you should use pg_edges here. 
_pg_edges, are both forward and backward links. 

I think you should use pg_links, 
which is only forward links.

Reply:
You are right, pg_links should work better.

Reply:
Sadly, the current test data does not account for _pg_links, we will have to revisit this in a different PR